webapp/src/js/services/format-data-record.js

```diff
@@ -55,35 +55,37 @@ angular
 
     const fieldsToHtml = function(
       settings,
+      doc,
       keys,
       labels,
-      data_record,
-      def,
-      locale
+      locale,
+      data,
+      def
     ) {
-      if (!def && data_record && data_record.form) {
-        def = getForm(settings, data_record.form);
+      if (!def && doc && doc.form) {
+        def = getForm(settings, doc.form);
       }
 
       if (_.isString(def)) {
         def = getForm(settings, def);
       }
 
+      if (!data) {
+        data = Object.assign({}, doc, doc.fields);
+      }
+
       const fields = {
         headers: [],
         data: [],
       };
 
-      const data = Object.assign({}, data_record, data_record.fields);
 
       _.forEach(keys, function(key) {
         if (_.isArray(key)) {
+          const result = fieldsToHtml(settings, doc, key[1], labels, locale, data[key[0]], def);
+          result.isArray = true;
+          fields.data.push(result);
           fields.headers.push({ head: titleize(key[0]) });
-          fields.data.push(
-            Object.assign(fieldsToHtml(key[1], labels, data[key[0]], def, locale), {
```

Comment:
As far as I can tell `fieldToHtml` was being passed entirely the wrong parameters... Now I'm not sure if the `key` is ever an array.

Reply:
YES! It can be an array and it seems to still not work properly.
This is the "sample" form I used (that has an Array key):
```
"NESTED": {
        "meta": {
          "code": "NESTED",
          "icon": "",
          "translation_key": "nested"
        },
        "fields": {
          "some.nested.field": {
            "labels": {
              "short": {
                "translation_key": "nested field"
              },
              "tiny": {
                "en": "nested"
              }
            },
            "position": 1,
            "flags": {},
            "type": "string",
            "required": false
          }
        },
        "use_sentinel": true
      },
```

Reply:
I had a look through and it's not a simple fix. I also assume the feature is not used because it's badly broken. Are you ok with me raising a new issue for this specific use case, so we can either remove support if it's not used, or get it to work properly if it is?

Reply:
Definitely ok with raising a separate issue!